libcodechecker/server/server.py

```diff
@@ -884,16 +886,42 @@ def start_server(config_directory, package_data, port, db_conn_string,
                      .format(root_file))
             root_sha = __make_root_file(root_file)
 
+        # Check whether configuration file exists, create an example if not.
+        server_cfg_file = os.path.join(config_directory, 'server_config.json')
+
+        if not os.path.exists(server_cfg_file):
+            # For backward compatibility reason if the session_config.json file
+            # exists we rename it to server_config.json.
+            session_cfg_file = os.path.join(config_directory,
+                                            'session_config.json')
+            example_cfg_file = os.path.join(os.environ['CC_PACKAGE_ROOT'],
+                                            'config', 'server_config.json')
+            if os.path.exists(session_cfg_file):
+                LOG.info("Renaming {0} to {1}. Please check the example "
+                         "configuration file ({2}) or the user guide for "
+                         "more information.".format(session_cfg_file,
+                                                    server_cfg_file,
+                                                    example_cfg_file))
+                os.rename(session_cfg_file, server_cfg_file)
+            else:
+                LOG.info("CodeChecker server's example configuration file "
+                         "created at " + server_cfg_file)
+                shutil.copyfile(example_cfg_file, server_cfg_file)
+
     try:
-        manager = session_manager.SessionManager(root_sha, force_auth)
+        manager = session_manager.SessionManager(
+            server_cfg_file,
```

Comment:
I am getting the following error when starting CodeChecker with an empty workspace:
```
UnboundLocalError: local variable 'server_cfg_file' referenced before assignment
```

Reply:
This is fixed. Damn you IDE for pasting code after an `else:` like it was meant to be in the else's statement body.